internal/extension/oop.go

```diff
@@ -158,3 +177,61 @@ func (p *OOPExtension) stopServer() error {
 	}
 	return nil
 }
+
+func (p *OOPExtension) monitorStderr(stderr io.ReadCloser, stopChan <-chan struct{}) {
+	scanner := bufio.NewScanner(stderr)
+	var lastReadTime time.Time
+
+	for {
+		select {
+		case <-stopChan:
+			// If the channel has been closed when the cmd has exited, we return
+			return
+		default:
+			//nolint:revive,staticcheck
+			if !lastReadTime.IsZero() && time.Since(lastReadTime) > 30*time.Second {
+				// We could check for liveness here
+			}
+
+			if scanner.Scan() {
+				lvl, text, err := parseStderr(scanner.Bytes())
+				p.logStderr(lvl, text, err)
+				lastReadTime = time.Now()
+			} else if err := scanner.Err(); err != nil {
+				p.logger.Error(err, "failed to read stderr")
+				return
+			}
+
+			// If this turns out to be causing busy-waiting/CPU spikes we could sleep for a brief time
+		}
+	}
+}
+
+func (p *OOPExtension) logStderr(logLevel int, logString string, err error) {
+	switch logLevel {
+	case 0:
+		p.logger.Info(logString)
+	default:
+		p.logger.Error(err, logString)
+	}
+}
+
+func parseStderr(logLine []byte) (logLevel int, logString string, err error) {
+	if len(logLine) == 0 {
+		return 1, "", fmt.Errorf("input byte slice is empty")
+	}
+
+	// Convert first byte to integer and validate log range
+	logLevel = int(logLine[0])
+	if logLevel < 0 || logLevel > 1 {
+		// At the moment only differencing from Info and Error
+		return 1, "", fmt.Errorf("first byte is not a valid log level range 0-1. log: %q", string(logLine))
+	}
```

Comment:
Would we log this line when the extension crashes? So that the first byte would be some ascii char?
If so, wondering if when the log level isn't `Info`, I'd log as `Error`... which btw, could we have that be the return type of `logLevel` rather than a cryptic `int`?

Reply:
If the extension crashes would be logged in line[ #99](https://github.com/Kuadrant/kuadrant-operator/pull/1261/files#diff-d747c090315f50f6d8cf8fb6aedd5da9452e910ab0bba5e473b57e757452e2ebR99), I could create a `logLevel` type to reflect the log level iota